src/pages/RFOX/components/Stake/StakeInput.tsx

```diff
@@ -346,60 +369,96 @@ export const StakeInput: React.FC<StakeInputProps & StakeRouteProps> = ({
   }, [])
 
   const handleSubmit = useCallback(() => {
-    if (!(stakingAssetAccountId && runeAddress && isValidStakingAmount)) return
+    if (!(assetAccountId && stakingAssetAccountId && runeAddress && isValidStakingAmount)) return
 
-    setConfirmedQuote({
+    const _confirmedQuote = {
       stakingAssetAccountId,
       stakingAssetId,
-      stakingAmountCryptoBaseUnit: toBaseUnit(amountCryptoPrecision, stakingAsset?.precision ?? 0),
-
+      stakingAmountCryptoBaseUnit: toBaseUnit(amountCryptoPrecision, asset?.precision ?? 0),
       runeAddress,
-    })
+    }
+
+    setConfirmedQuote(_confirmedQuote)
+
+    if (isBridgeRequired) {
+      const bridgeQuote: RfoxBridgeQuote = {
+        sellAssetId: assetId,
+        buyAssetId: stakingAssetId,
+        bridgeAmountCryptoBaseUnit: toBaseUnit(amountCryptoPrecision, asset?.precision ?? 0),
```

Comment:
also here, missing asset paranoia should handle

Reply:
See https://github.com/shapeshift/web/pull/7064#discussion_r1630923837

Reply:
three birds one stone in 62532f132f

1. Renamed all related `asset` vars to `selectedAsset` for clarity
2. Sane naming makes everything saner, self-reviewed the diff after those renames and noticed a few occurences of selected asset being used in place of staking asset and vice-versa
3. Added paranoia early returns for `!asset`

Reply:
And paranoia retest here:

- Not eough ETH.ARB for gas with FOX.ETH selected

<img width="523" alt="image" src="https://github.com/shapeshift/web/assets/17035424/71197634-028d-47d0-9641-67f64964bf95">

- Not eough ETH.ARB for gas with FOX.ARB selected

<img width="519" alt="image" src="https://github.com/shapeshift/web/assets/17035424/50f5f0e0-57e2-4dd7-836b-a941fb4172a5">

- Enough ETH.ARB for gas with FOX.ETH selected

<img width="523" alt="image" src="https://github.com/shapeshift/web/assets/17035424/98cbe7be-8c5d-49de-8b44-84e65eed4bec">

- Enough ETH.ARB for gas with FOX.ARB selected

<img width="529" alt="image" src="https://github.com/shapeshift/web/assets/17035424/68fa942b-2998-48f4-9671-75b73a2093a9">

- Stake ack

<img width="511" alt="image" src="https://github.com/shapeshift/web/assets/17035424/74a5f7f5-ad97-41eb-b270-b617f8a224eb">

- Bridge ack

<img width="505" alt="image" src="https://github.com/shapeshift/web/assets/17035424/236cbdc1-d02f-4807-b39a-896cb08d115d">

- Bridge quote still using the correct sell and buy assets

<img width="501" alt="image" src="https://github.com/shapeshift/web/assets/17035424/af3b00ad-c37b-4106-b8b9-85c8a530d419">